test/Core/Utility/PathTests.re

```diff
@@ -0,0 +1,73 @@
+open TestFramework;
+
+open Oni_Core.Utility;
+
+describe("Path", ({test, _}) => {
+  test("hasTrailingSlash", ({expect}) => {
+    let cases = [
+      ("", false),
+      ("a", false),
+      ("a\\", true),
+      ("a/", true),
+      ("\\", true),
+      ("/", true),
+    ];
+
+    let runCase = ((path, expectedResult)) => {
+      expect.bool(Path.hasTrailingSeparator(path)).toBe(expectedResult);
+    };
+
+    List.iter(runCase, cases);
+  });
+
+  test("trimTrailingSeparator", ({expect}) => {
+    let cases = [
+      ("", ""),
+      ("\\", ""),
+      ("/", ""),
+      ("a/", "a"),
+      ("b\\", "b"),
+      ("a/b/", "a/b"),
+      ("a\\b\\", "a\\b"),
+    ];
+
+    let runCase = ((path, expectedResult)) => {
+      expect.string(Path.trimTrailingSeparator(path)).toEqual(
+        expectedResult,
+      );
+    };
+
+    List.iter(runCase, cases);
+  });
+
+  test("toRelative", ({expect}) => {
+    // POSIX-style paths
+    expect.string(
+      Path.toRelative(~base="/Applications", "/Applications/Onivim2.app"),
+    ).
+      toEqual(
+      "Onivim2.app",
+    );
+
+    expect.string(
+      Path.toRelative(~base="/Applications/", "/Applications/Onivim2.app"),
+    ).
+      toEqual(
+      "Onivim2.app",
+    );
+
+    expect.string(Path.toRelative(~base="/", "/Applications/Onivim2.app")).
+      toEqual(
+      "Applications/Onivim2.app",
+    );
+
+    // Windows-style paths
+
+    expect.string(Path.toRelative(~base="D:", "D:\\Onivim2")).toEqual(
+      "Onivim2",
+    );
```

Comment:
How does this pass CI? It fails for me locally because `Filename.dir_sep` is `/`, so it tried to remove `D:/`, which obviously doesn't work, so it returns the original string. This test should fail on anything other than Windows.

Reply:
Hmm, good question, I'm looking into this now. I figured since it was green it was good, but I can repro the failure locally too.

Reply:
I added a new test project as part of that change, and I forgot that there is a _separate_ test project I needed to add it for CI. So `esy @test run` fails, but `esy @test run-ci` passes, since it wasn't running that test...

Opened a PR here: https://github.com/onivim/oni2/pull/1296

Reply:
Sorry you hit this @glennsl ! Would've been confusing on top of the 4.08 changes... hope it didn't take too much time from you.

Reply:
No worries. The change was small and very recent, so didn't take much to figure out at all.